src_c/rect.c

```diff
@@ -572,21 +572,36 @@ pg_rect_unionall_ip(pgRectObject *self, PyObject *args)
 }
 
 static PyObject *
-pg_rect_collidepoint(pgRectObject *self, PyObject *args)
+pg_rect_collidepoint(pgRectObject *self, PyObject *const *args,
+                     Py_ssize_t nargs)
 {
+    SDL_Rect srect = self->r;
     int x = 0, y = 0;
     int inside;
+    if (nargs == 1) {
+        if (!pg_TwoIntsFromObj(args[0], &x, &y))
+            return RAISE(PyExc_TypeError, "argument must contain two numbers");
+    }
+    else if (nargs == 2) {
+        x = PyLong_AsLong(args[0]);
+        if (PyErr_Occurred())
+            return NULL;
 
-    if (!pg_TwoIntsFromObj(args, &x, &y)) {
-        return RAISE(PyExc_TypeError, "argument must contain two numbers");
+        y = PyLong_AsLong(args[1]);
```

Comment:
As mentioned on discord, using `PyLong_AsLong` can potentially break the usage of this function with floats. (`pg_TwoIntsFromObj` handles floats too but truncates them)